src/server/controllers/Users.ts

```diff
@@ -1,23 +1,34 @@
-const users: User[] = [];
+const users: UserType[] = [];
 
-// Join user to chat
-export const userJoin = ({
-  id, username, room
-}: {
-  id: string, username: string, room: string }
-): User => {
-  const user = { id, username, room };
-  users.push(user);
-  return user;
-};
+// TODO: finish moving User to model
+export class User {
+  id: string;
+  username: string;
+  room: string;
+  isLeader: boolean;
+  
+  constructor(user: UserType) {
+    const { id, username, room } = user;
+    this.id = id;
+    this.username = username;
+    this.room = room;
+    // if room exists ( === already has leader) - false
+    // if room doesn't exist - true
+    this.isLeader = false; // TODO: apply logic
+    users.push(this);
+
+    // if not username or room - return error ?
```

Comment:
it's mandatory in type so no need

Reply:
need checking in Game class. to only create User if the username is not taken. if username exists send back error